### PR TITLE
Rename hydration terms

### PR DESCRIPTION
**Pull request by gnawf**
image

### CODE REVIEW

File: lib/src/main/java/graphql/nadel/definition/hydration/NadelHydrationDefinition.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ class NadelHydrationDefinition(
                 "This allows you to hydrate new values into fields"
                 directive @hydrated(
                     "The target service"
-                    service: String!
+                    service: String
```
Comment (Juliano-Prado) [inline lib/src/main/java/graphql/nadel/definition/hydration/NadelHydrationDefinition.kt:36]: is it right, service now can be null?

Comment (gnawf) [reply lib/src/main/java/graphql/nadel/definition/hydration/NadelHydrationDefinition.kt:36]: Yeah sorry I kind of snuck that into this PR. That field is going to be removed in the future. It's going to be inferred from the service it's declared in.

File: docs/Primer.MD [hunk 2]
```diff
@@ -249,9 +249,9 @@ Is transformed to:
 data class NadelHydrationFieldInstruction(
     override val location: FieldCoordinates, // User.friend
     val hydratedFieldDef: GraphQLFieldDefinition, // Schema definition for User.friend
-    val actorService: Service, // Users service i.e. service owning users.friendById
-    val queryPathToActorField: NadelQueryPath, // users.friendById
-    val actorFieldDef: GraphQLFieldDefinition, // Schema definition for users.friendById
+    val backingService: Service, // Users service i.e. service owning users.friendById
+    val queryPathToBackingField: NadelQueryPath, // users.friendById
+    val backingFieldDef: GraphQLFieldDefinition, // Schema definition for users.friendById
     val actorInputValueDefs: List<NadelHydrationActorInputDef>, // [{ name: "ids" value: "$source.friendId" }]
```
Comment (Juliano-Prado) [inline docs/Primer.MD:255]: what about this actorInput? Does not make sense to rename to backingInputValueDefs?

Comment (gnawf) [reply docs/Primer.MD]: Updated, thanks for pointing that out

**Review by Juliano-Prado — APPROVED** [review]
LGTM

**Pull request merged**